src/commands/utils/azureAccount.ts

```diff
@@ -144,39 +184,43 @@ function getMicrosoftGraphClientBaseUrl(environment: Environment): string {
     // They do not contain the MS Graph endpoints, whose values are here:
     // https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/d365ab1d68f90f2c38c67a5a7c7fe54acfc2584e/src/Constants.ts#L28
     switch (environment.name) {
-        case 'AzureChinaCloud':
+        case "AzureChinaCloud":
             return "https://microsoftgraph.chinacloudapi.cn";
-        case 'AzureUSGovernment':
+        case "AzureUSGovernment":
             return "https://graph.microsoft.us";
-        case 'AzureGermanCloud':
+        case "AzureGermanCloud":
             return "https://graph.microsoft.de";
     }
 
     return "https://graph.microsoft.com";
 }
 
-async function getSubscriptionAccess(credential: TokenCredential, subscription: Subscription, spInfo: ServicePrincipalInfo): Promise<Errorable<SubscriptionAccessResult>> {
+async function getSubscriptionAccess(
+    credential: TokenCredential,
+    subscription: Subscription,
+    spInfo: ServicePrincipalInfo,
+): Promise<Errorable<SubscriptionAccessResult>> {
     if (!subscription.subscriptionId) {
-        return { succeeded: true, result: {subscription, hasRoleAssignment: false} };
+        return { succeeded: true, result: { subscription, hasRoleAssignment: false } };
     }
 
     const client = new AuthorizationManagementClient(credential, subscription.subscriptionId);
     let roleAssignments: RoleAssignmentsListResponse;
     try {
-        roleAssignments = await client.roleAssignments.list({filter: `principalId eq '${spInfo.id}'`});
+        roleAssignments = await client.roleAssignments.list({ filter: `principalId eq '${spInfo.id}'` });
     } catch (e) {
         if (isUnauthorizedError(e)) {
-            return { succeeded: true, result: {subscription, hasRoleAssignment: false} };
+            return { succeeded: true, result: { subscription, hasRoleAssignment: false } };
         }
 
         return { succeeded: false, error: getErrorMessage(e) };
     }
 
     // The service principal needs *some* permissions in the subscription, but Contributor is not
     // necessarily required. See: https://azure.github.io/azure-service-operator/#installation
-    return { succeeded: true, result: {subscription, hasRoleAssignment: roleAssignments.length > 0}};
+    return { succeeded: true, result: { subscription, hasRoleAssignment: roleAssignments.length > 0 } };
 }
 
-function isUnauthorizedError(e: any): boolean {
-    return e.code === 'AuthorizationFailed' && e.statusCode === 403;
+function isUnauthorizedError(e: unknown): boolean {
+    return isObject(e) && "code" in e && "statusCode" in e && e.code === "AuthorizationFailed" && e.statusCode === 403;
```

Comment:
I don't think this will behave differently in practice, because errors are _usually_ objects (especially from trusted libraries), but this adds and extra check just to make sure.